cypress-tests/cypress/e2e/v2/projectResources.cy.ts

```diff
@@ -0,0 +1,214 @@
+        deleteProjectFromAPIV2(projectIdentifier);
+      }
+    });
+  });
+
+  const visitCurrentProject = () => {
+    // There should a link on the dashboard for a newly created project
+    cy.visit("/v2");
+    cy.getDataCy("dashboard-project-list")
+      .get(
+        `a[href*="/${projectIdentifier.namespace}/${projectIdentifier.slug}"]`,
+      )
+      .click();
+    cy.getDataCy("project-name").should("contain", projectName);
+  };
+
+  it("Work with data", () => {
+    const name = `giab-${getRandomString()}`;
+
+    // Add data connector
+    visitCurrentProject();
+    cy.getDataCy("add-data-connector").click();
+    cy.getDataCy("project-data-controller-mode-create").click();
+    cy.getDataCy("data-connector-edit-next-button").should("be.disabled");
+    cy.getDataCy("data-storage-s3").click();
+    cy.getDataCy("data-provider-AWS").click();
+    cy.getDataCy("data-connector-edit-next-button").click();
+
+    cy.getDataCy("data-connector-source-path").should("be.empty").type("giab");
+    cy.getDataCy("test-data-connector-button").click();
+    cy.getDataCy("cloud-storage-connection-success").should("be.visible");
+    cy.getDataCy("add-data-connector-continue-button").click();
+
+    cy.getDataCy("data-connector-name-input").should("be.empty").type(name);
+    cy.getDataCy("data-connector-slug-input").should("have.value", name);
+    cy.getDataCy("data-connector-mount-input").should("have.value", name);
+    cy.getDataCy("data-connector-edit-update-button").click();
+    cy.getDataCy("data-connector-edit-success").should("be.visible");
+    cy.getDataCy("data-connector-edit-close-button").click();
+
+    // ? Currently, data connectors newly linked might not appear immediately
+    visitCurrentProject();
+    cy.getDataCy("data-connector-box")
+      .find(`[data-cy=data-connector-name]`)
+      .contains(name);
+
+    // Edit data connector
+    const newName = `${name} edited`;
+
+    cy.getDataCy("data-connector-box")
+      .find(`[data-cy=data-connector-name]`)
+      .contains(name)
+      .click();
+    cy.getDataCy("data-connector-edit").click();
+    cy.getDataCy("test-data-connector-button").click();
+    cy.getDataCy("add-data-connector-continue-button").click();
+
+    cy.getDataCy("data-connector-name-input")
+      .should("have.value", name)
+      .clear()
+      .type(newName);
+    cy.getDataCy("data-connector-slug-input").should("have.value", name);
+    cy.getDataCy("data-connector-mount-input").should("have.value", name);
+    cy.getDataCy("data-connector-edit-update-button").click();
+    cy.getDataCy("data-connector-edit-success").should("be.visible");
+    cy.getDataCy("data-connector-edit-close-button").click();
+
+    // ? Currently, data connectors newly linked might not appear immediately
+    visitCurrentProject();
+    cy.getDataCy("data-connector-box")
+      .find(`[data-cy=data-connector-name]`)
+      .contains(newName);
+  });
+
+  it("Work with code", () => {
```

Comment:
Same as the rest  the title is too general making it sound like we try a lot more in the tests.

```suggestion
  it("Add and modify code repositories", () => {
```